pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/ConsumerCoordinator.java

```diff
@@ -299,26 +227,36 @@ AtomicBoolean getFirstTransitionProcessed() {
     return _firstTransitionProcessed;
   }
 
-  // this should not be used outside of tests.
   @VisibleForTesting
-  int getMaxSegmentSeqNumLoaded() {
-    return _maxSegmentSeqNumRegistered;
+  int getMaxSequenceNumberRegistered() {
+    return _maxSequenceNumberRegistered;
   }
 
   @VisibleForTesting
-  boolean isSegmentAlreadyConsumed(String currSegmentName) {
-    SegmentZKMetadata segmentZKMetadata = _realtimeTableDataManager.fetchZKMetadata(currSegmentName);
+  void checkSegmentZKMetadataStatus(String segmentName)
+      throws ShouldNotConsumeException {
+    SegmentZKMetadata segmentZKMetadata = _realtimeTableDataManager.fetchZKMetadataNullable(segmentName);
```

Comment:
This is a bug fix. Existing code directly throw exception when segment is deleted